lib/vagrant-libvirt/config.rb

```diff
@@ -659,10 +659,26 @@ def finalize!
         @management_network_domain = nil if @management_network_domain == UNSET_VALUE
         @system_uri      = 'qemu:///system' if @system_uri == UNSET_VALUE
 
+        # Default to false for qemu_use_session
         @qemu_use_session = false if @qemu_use_session == UNSET_VALUE
 
-        # generate a URI if none is supplied
-        @uri = _generate_uri if @uri == UNSET_VALUE
+        # If uri is not set then:
+        #   - set based on LIBVIRT_DEFAULT_URI in env if exists
+        #   - OR generate it from scratch
+        if @uri == UNSET_VALUE
+          if ENV['LIBVIRT_DEFAULT_URI']
```

Comment:
Probably going to have to do a little more testing here, as `LIBVIRT_DEFAULT_URI= ...` is sufficent not to return nil resulting in entering this block but setting the uri to empty which would be undesirable.